### DISCUSSION TITLE
Self hosted runner SSL issue

### DISCUSSION THREAD

**Discussion by ghost** [community/community#26728] [Actions]
<p>Hi,<br>
I am running self hosted runner</p>
<p>The workflow starts correctly, checkout works ok but when going to<br>
actions/setup-java@v2</p>
<p>I get the error below</p>
<blockquote>
<p>unable to verify the first certificate</p>
</blockquote>
<p>Thanks for your help</p>

**Comment by jsoref** [comment]
<p>You should file an issue here:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/actions/setup-java/issues">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/actions/setup-java/issues" target="_blank" rel="noopener">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/2119212/181099448-796a6ce7-2e5d-4b18-b7b2-debfd7d769cc.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/6/f6c6939e5fb9adadd4c3376f3e34ab22d9194edd_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/f/6/f6c6939e5fb9adadd4c3376f3e34ab22d9194edd_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/f/6/f6c6939e5fb9adadd4c3376f3e34ab22d9194edd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/6/f6c6939e5fb9adadd4c3376f3e34ab22d9194edd_2_10x10.png"></div>

<h3><a href="https://github.com/actions/setup-java/issues" target="_blank" rel="noopener">Issues · actions/setup-java</a></h3>

  <p>Set up your GitHub Actions workflow with a specific version of Java - Issues · actions/setup-java</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by gitstua** [comment]
<p>It’s not an ideal security situation to do this permanently but you could see if disabling TLS verification fixes your problem: <a href="https://docs.github.com/en/actions/hosting-your-own-runners/monitoring-and-troubleshooting-self-hosted-runners#disabling-tls-certificate-verification" class="inline-onebox" rel="noopener nofollow ugc">Monitoring and troubleshooting self-hosted runners - GitHub Docs</a></p>
<p>Do your self hosted runners connect to GitHub via a corporate proxy? some of those replace the certificate with a “man in the middle attack” hence the error</p>
<p>Another thought is to make sure the runner is up to date - there was a certificate change for <a href="http://github.com" rel="noopener nofollow ugc">github.com</a> recently: <a href="https://www.ssllabs.com/ssltest/analyze.html?d=www.github.com#:~:text=05189a54ebe8c7e903e0ab0d925545de-,Valid%20from,Tue%2C%2015,-Mar%202022%2000" class="inline-onebox" rel="noopener nofollow ugc">SSL Server Test: www.github.com (Powered by Qualys SSL Labs)</a></p>

**Comment by ghost** [comment]
<p><a class="mention" href="/gitstua">@gitstua</a> I asked the proxy guys to check. I suspect it is blocked by our proxy and need to be whitelisted.<br>
Thanks</p>

**Comment by ghost** [comment]
<p>The url have been whitelisted by our proxy team but it still doesn’t work.</p>
<p>Proxy is set in the runner, the command env proves it.</p>
<p>Is there a possibility that the proxy settings are not taken for the setup part ?</p>
<p>When doing curl on urls, it works fine :<br>
<a href="https://github.com/actions/python-versions/releases/download/3.10.4-2034678717/python-3.10.4-linux-20.04-x64.tar.gz" rel="noopener nofollow ugc">https://github.com/actions/python-versions/releases/download/3.10.4-2034678717/python-3.10.4-linux-20.04-x64.tar.gz</a> -o python-source.tar.gz<br>
or<br>
<a href="https://pipelines.actions.githubusercontent.com/_apis/health" rel="noopener nofollow ugc">https://pipelines.actions.githubusercontent.com/_apis/health</a><br>
or<br>
<a href="https://api.adoptopenjdk.net" rel="noopener nofollow ugc">https://api.adoptopenjdk.net</a></p>

**Comment by gitstua** [comment]
<p>Sounds frustrating to troubleshoot. Sounds like you should follow up through <a href="https://support.github.com/" rel="noopener nofollow ugc">https://support.github.com/</a></p>

**Comment by ghost** [comment, marked as answer]
<p>Hi, finally I had to set :</p>
<pre><code class="lang-auto">env:
     - name: SSL_CERT_DIR
       value: "/etc/ssl/certs"
</code></pre>
<p>In the runner to make it work.<br>
Thanks</p>

**Reply by bantl23** [reply]
@RaphaelPerrinPost Where in the runner did you set this?

I tried added it in .github/workflows/action.yml but that didn't work. I'm assuming you mean some setting on the runner itself?

**Reply by ghost** [reply]
@bantl23 in the runner deployment:

>       . . .      
>        containers: 
>         - name: runner
>           image: "summerwind/actions-runner:latest"
>           env:
>           - name: SSL_CERT_DIR
>             value: "/etc/ssl/certs"
>         . . . 

 
I hope this helps.

**Reply by bantl23** [reply]
I see your runner is hosted in kubernetes, I missed that. Thank you!
